Corrupted TSM file causes panic during startup, doesn't indicate which file is root cause

From commit 821525017fcc0610ee44549f13e3caa51363e879:

I was recovering from a disaster where shards on disk were out of sync with what the meta store knew about, and I was exporting TSM and WAL files back to line protocol with `influx_inspect` then importing them back with `influx -import`.

Something during import crashed the server, but there were so many logs that it far exceeded my scrollback buffer and I was unable to diagnose it. After that crash, the server panics on every restart with:

```
panic: not ordered: 964 1457275119000000000 >= 1457275119000000000

goroutine 9642 [running]:
panic(0xb17920, 0xc82b7f36a0)
	/usr/local/go/src/runtime/panic.go:481 +0x3e6
github.com/influxdata/influxdb/tsdb/engine/tsm1.Values.assertOrdered(0xc8326fc000, 0x74c, 0xe00)
	/root/go/src/github.com/influxdata/influxdb/tsdb/engine/tsm1/encoding.gen.go:39 +0x2fc
github.com/influxdata/influxdb/tsdb/engine/tsm1.(*tsmKeyIterator).chunk(0xc8264d2700, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0)
	/root/go/src/github.com/influxdata/influxdb/tsdb/engine/tsm1/compact.go:1026 +0x70
github.com/influxdata/influxdb/tsdb/engine/tsm1.(*tsmKeyIterator).combine(0xc8264d2700, 0xf60f01, 0x0, 0x0, 0x0)
	/root/go/src/github.com/influxdata/influxdb/tsdb/engine/tsm1/compact.go:953 +0x16f
github.com/influxdata/influxdb/tsdb/engine/tsm1.(*tsmKeyIterator).merge(0xc8264d2700)
	/root/go/src/github.com/influxdata/influxdb/tsdb/engine/tsm1/compact.go:901 +0x12f
github.com/influxdata/influxdb/tsdb/engine/tsm1.(*tsmKeyIterator).Next(0xc8264d2700, 0xc8000d5912)
	/root/go/src/github.com/influxdata/influxdb/tsdb/engine/tsm1/compact.go:777 +0xe7
github.com/influxdata/influxdb/tsdb/engine/tsm1.(*Compactor).write(0xc82d3cb5e0, 0xc82babbcc0, 0x41, 0x7fa401331820, 0xc8264d2700, 0x0, 0x0)
	/root/go/src/github.com/influxdata/influxdb/tsdb/engine/tsm1/compact.go:607 +0x53b
github.com/influxdata/influxdb/tsdb/engine/tsm1.(*Compactor).writeNewFiles(0xc82d3cb5e0, 0x14, 0x3, 0x7fa401331820, 0xc8264d2700, 0x0, 0x0, 0x0, 0x0, 0x0)
	/root/go/src/github.com/influxdata/influxdb/tsdb/engine/tsm1/compact.go:554 +0x3a1
github.com/influxdata/influxdb/tsdb/engine/tsm1.(*Compactor).compact(0xc82d3cb5e0, 0x484c01, 0xc82ce33780, 0x2, 0x2, 0x0, 0x0, 0x0, 0x0, 0x0)
	/root/go/src/github.com/influxdata/influxdb/tsdb/engine/tsm1/compact.go:513 +0x4fe
github.com/influxdata/influxdb/tsdb/engine/tsm1.(*Compactor).CompactFast(0xc82d3cb5e0, 0xc82ce33780, 0x2, 0x2, 0x0, 0x0, 0x0, 0x0, 0x0)
	/root/go/src/github.com/influxdata/influxdb/tsdb/engine/tsm1/compact.go:539 +0xfd
github.com/influxdata/influxdb/tsdb/engine/tsm1.(*Engine).compactTSMLevel.func1(0xc82d82dce0, 0xc826839140, 0x2, 0x23746c7561666501, 0x0, 0xc82ce33780, 0x2, 0x2)
	/root/go/src/github.com/influxdata/influxdb/tsdb/engine/tsm1/engine.go:764 +0x600
created by github.com/influxdata/influxdb/tsdb/engine/tsm1.(*Engine).compactTSMLevel
	/root/go/src/github.com/influxdata/influxdb/tsdb/engine/tsm1/engine.go:790 +0x265
```

Because it's a panic from a compaction running in a goroutine, while the server is still starting, the lines immediately before the panic are different on every restart.  The panic does not include information about what TSM file is corrupt; and this server has over 2000 .tsm files, so it's no small task to figure out which one is causing the panic. 

At a minimum, this panic should include the name of the TSM file that resulted in the panic, so that the operator can move the file out of the way.

--- a/tsdb/engine/tsm1/compact.go
+++ b/tsdb/engine/tsm1/compact.go
@@ -1158,8 +1158,6 @@ func (k *tsmKeyIterator) combine(dedup bool) blocks {
 }
 
 func (k *tsmKeyIterator) chunk(dst blocks) blocks {
-	k.mergedValues.assertOrdered()
-
 	for len(k.mergedValues) > k.size {
 		values := k.mergedValues[:k.size]
 		cb, err := Values(values).Encode(nil)

